datasketch/experimental/aio/lsh.py

```diff
@@ -0,0 +1,389 @@
+from datasketch.storage import unordered_storage, _random_name
+
+
+class AsyncMinHashLSH(object):
+    """
+    Asynchronous MinHashLSH index.
+
+    :param float threshold: see :class:`datasketch.MinHashLSH`.
+    :param int num_perm: see :class:`datasketch.MinHashLSH`.
+    :param weights: see :class:`datasketch.MinHashLSH`.
+    :type weights: tuple(float, float)
+    :param tuple params: see :class:`datasketch.MinHashLSH`.
+    :param dict storage_config: New types of storage service (aioredis, aiomongo) to use for storing hashtables and keys are implemented. If storage_config is None aioredis storage will be used.
+    :param bool prepickle: for redis type storage use bytes as keys.
+    :param byte name: basename. Useful if you want pickle LSH class and spread it into different workers.
+
+    Example:
+        There is two ways to create a AsyncMinHash index.
+            1. Usual way:
+
+                .. code-block:: python
+
+                    from datasketch.experimental.aio import AsyncMinHashLSH
+
+                    _storage_config_redis = {'type': 'aioredis', 'redis': {'host': 'localhost', 'port': 6379}}
+
+                    lsh = await AsyncMinHashLSH(storage_config=_storage_config_redis, threshold=0.5, num_perm=16)
+                    m1 = MinHash(16)
+                    m1.update('a'.encode('utf8'))
+                    m2 = MinHash(16)
+                    m2.update('b'.encode('utf8'))
+                    await lsh.insert('a', m1)
+                    await lsh.insert('b', m2)
+                    print(await lsh.query(m1))
+                    print(await lsh.query(m2))
+                    lsh.close()
+
+            2. Context Manager style:
+
+                .. code-block:: python
+
+                    from datasketch.experimental.aio import AsyncMinHashLSH
+
+                    _storage_config_redis = {'type': 'aioredis', 'redis': {'host': 'localhost', 'port': 6379}}
+
+                    aio with AsyncMinHashLSH(storage_config=_storage_config_redis, threshold=0.5, num_perm=16) as lsh:
+                        m1 = MinHash(16)
+                        m1.update('a'.encode('utf8'))
+                        m2 = MinHash(16)
+                        m2.update('b'.encode('utf8'))
+                        await lsh.insert('a', m1)
+                        await lsh.insert('b', m2)
+                        print(await lsh.query(m1))
+                        print(await lsh.query(m2))
+
+    .. note::
+        * The module supports Python version >=3.6, and is currently experimental. So the interface may change slightly in the future.
+        * For main functionality of LSH algorithm see :class:`datasketch.MinHashLSH`.
+        * For additional information see :ref:`minhash_lsh_at_scale` and :ref:`minhash_lsh_async`
+    """
+
+    def __init__(self, threshold=0.9, num_perm=128, weights=(0.5, 0.5),
+                 params=None, storage_config=None, prepickle=None, name=None):
```

Comment:
`name` is base name, so it should be called `base_name`.

Reply:
yes. you're right